wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+use CRM_Pets_ExtensionUtil as E;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+
+		// Access the subtype
+		$subtype = $decodedData['Eck_Collection_Camp1'][0]['fields']['subtype'] ?? null;
+
+		// Check if the subtype is 4 or 5
+		if ($subtype != 4 && $subtype != 5) {
```

Comment:
4 and 5 seems to be hardcoded IDs of the subtypes. We don't want to use them. We want to use names that do not change.

Reply:
We've used the Subtype field on the FormBuilder form, but when the form status changes, the logs only show the subtype ID. That's why I used the ID for conditional checks. 
I have added some mappings between the ids and label  to use the names instead of the ids
<img width="813" alt="image" src="https://github.com/user-attachments/assets/20cb83cf-be2e-4def-a6a9-7195588e8c72">